### PR TITLE
test implementation for controllers user and post

### PR DESCRIPTION
**Pull request by alexansaa**
In this milestone, the following features have been implemented:
- Test files for UsersController and PostsControllers
- Tests have been implemented for methods within Users and Posts Controllers (index and show )
- Tests have been implemented for checking the correctness of the status response
- Tests have been implemented for checking the correctness of the template renderization
- Tests have been implemented for checking that the response body includes placeholder text

### CODE REVIEW

File: spec/controller/post_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  describe 'GET #index' do
+    it 'returns http success' do
+      get :index, params: { user_id: 1 }
+      expect(response).to be_successful
+    end
+
+    it 'renders the index view' do
+      get :index, params: { user_id: 1 }
+      assert 'title', 'Here is a list of posts for User Id: 1'
+    end
```
Comment (mckent05) [inline spec/controller/post_controller_spec.rb]: - Good job done so far, but according to the project requirements, you are to write tests that ensure that the correct template is rendered, just as you did in the `post#show` test below. Please fix this 😄

File: spec/controller/user_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+RSpec.describe UsersController, type: :controller do
+  describe 'GET #index' do
+    it 'returns http success' do
+      get :index
+      expect(response).to be_successful
+    end
+
+    it 'renders the index view' do
+      get :index
+      assert 'title', 'Here is a list of users, as the main page'
```
Comment (mckent05) [inline spec/controller/user_controller_spec.rb]: - Good job done so far, but according to the project requirements, you are to write tests that ensure that the correct template is rendered, just as you did in the `user#show` test below. Please fix this 😄

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @alexansaa, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job adding specs for your controllers 👍 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @alexansaa, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**